src/mds/Server.cc

```diff
@@ -362,8 +362,9 @@ void Server::dispatch(const cref_t<Message> &m)
     handle_peer_request(ref_cast<MMDSPeerRequest>(m));
     return;
   default:
-    derr << "server unknown message " << m->get_type() << dendl;
-    ceph_abort_msg("server unknown message");  
```

Comment:
How to identify that circumstance where an MDS peer is sending an unknown message to this MDS?

Reply:
The message has a reference to the connection, and the connection has a (secure) check of the entity on the other side.
So look at m->con->get_peer_type().

Reply:
You can also use `m->get_source().is_client()` to check whether it's peer is client or not. If it's then you can do the `evict_client()`.

For all the others I am afraid we have no choice other than just dropping them, because we even don't know how to tell the peers what has happened. What type of reply struct should we use ?

And instead of just silently dropping it, we'd better report a warning to mon, at least admin should know what has happened.

Reply:
Thanks @lxbsz, I've used it in one of the conditions :)

Reply:
Okay, sound reasonable.

Reply:
@batrick Thanks! I'll incorporate the changes.

Reply:
@gregsfortytwo Thanks greg, I've made the change

Reply:
@dparmar18 Is this check even implemented?

Reply:
@vshankar After some discussion with patrick, it was decided that we don't need to patch `Server::dispatch` since it indicates a bug in `MDSRank`. Check this out https://github.com/ceph/ceph/pull/47080#discussion_r926164349